File: scrimage-core/src/main/scala/com/sksamuel/scrimage/canvas/drawable/DrawableImage.scala
package com.sksamuel.scrimage.canvas.drawable

import java.awt.Graphics2D

import com.sksamuel.scrimage.ImmutableImage
import com.sksamuel.scrimage.canvas.{Drawable, GraphicsContext}

case class DrawableImage(imageToDraw: ImmutableImage, x: Int, y: Int, context: GraphicsContext) extends Drawable {
  def draw(g: Graphics2D): Unit = g.drawImage(imageToDraw.awt, x, y, null)
}

File: scrimage-core/src/main/java/com/sksamuel/scrimage/canvas/drawables/DrawableImage.java
package com.sksamuel.scrimage.canvas.drawables;

import com.sksamuel.scrimage.ImmutableImage;
import com.sksamuel.scrimage.canvas.Drawable;
import com.sksamuel.scrimage.canvas.GraphicsContext;
import com.sksamuel.scrimage.canvas.RichGraphics2D;

public class DrawableImage implements Drawable {

    private final ImmutableImage image;
    private final int x;
    private final int y;
    private final GraphicsContext context;

    public DrawableImage(ImmutableImage image, int x, int y, GraphicsContext context) {
        this.image = image;
        this.x = x;
        this.y = y;
        this.context = context;
    }

    public DrawableImage(ImmutableImage image, int x, int y) {
        this.image = image;
        this.x = x;
        this.y = y;
        this.context = GraphicsContext.identity();
    }


    @Override
    public void draw(RichGraphics2D g) {
        g.drawImage(image.awt(), x, y, null);
    }

    @Override
    public GraphicsContext context() {
        return context;
    }
}
